.github/workflows/deploy.yml

```diff
@@ -0,0 +1,36 @@
+name: deploy
+
+on:
+  pull_request:
+    branches: [main]
```

Comment:
This runs for all pull requests where in practice you would only want this to run when CI has passed.